### PR TITLE
feat(needed-items): search pooled quest objectives by any accepted item

### LINKED ISSUE #812: [Enhancement]: Show items from "any of X" / pool quest objectives in Needed Items
### Which area of TarkovTracker is affected?

Needed items

### Current Behavior

Currently, the Needed Items page only shows items for quests that require a specific, single item. Tasks where you can turn in any items from a pool (like Pets Won't Need It, which accepts any 56 medical items) don't show up here at all. For example, if I search for "Augmentin" or "antibiotic", nothing shows up for Pets Won't Need It, even though Augmentin is a valid turn-in item for that quest.

### Proposed Enhancement

It would be great if valid turn-in items from these pooled quests were also searchable and visible on the Needed Items page. To avoid confusion, they could have a small tag or note indicating that they belong to an optional/shared pool (e.g. "Any of 56" or "Pool objective"), rather than looking like an absolute hard requirement.

### Benefits

Players won't accidentally sell, consume, or ignore valid quest turn-in items sitting in their stash or found in raid.

Search on the Needed Items page will reflect actual quest needs much more accurately.

Reduces the need to check third-party wikis or the full quest page just to remember what medical items or generic turn-ins are currently useful.

### What does this prevent you from doing?

I can't use the Needed Items search to check if an item I just found is useful for one of my active quests if that quest uses a flexible item pool. It forces me to manually navigate to the quest itself or double-check the wiki.

### How often does this affect you?

Rarely

### Who would benefit from this change?

Every time I have quests with broad category requirements active (especially medical turn-in quests like Pets Won't Need It or barter-item pool handovers). It affects regular players managing stash space and deciding what to keep from raids.

### Screenshots/Examples

Image

### Discord Handle (Optional)

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by DysektAI**
## Summary

Resolves #812.

Pooled "any of these" quest objectives (`acceptedItems` with more than one entry, e.g. Pets Won't Need It — any 1 of 56 medical items) previously only surfaced their primary/cycled item on the Needed Items page. Searching for any other valid turn-in item (e.g. "Augmentin") returned nothing, forcing users to check the task page or wiki.

### Changes

- **Search matches any accepted item** (`useNeededItems.ts`): pooled task objectives now match when the query fuzzy-matches any accepted item name/short name, not just the primary item. Shared helpers `itemMatchesQuery` / `findAcceptedItemMatchIndex` added to `neededItemFilters.ts` and reused by the filter, the display pin, and the grouped view.
- **Pinned display while searching** (`NeededItem.vue`, `useCyclingItem.ts`): when the search matches one of a pooled objective's accepted items, the list/grid row or card pins to that item (rotation pauses) so users see exactly what matched. The existing "Any of N" accepted-items popover keeps pool membership explicit, satisfying the "small tag" ask from the issue. `useCyclingItem` gains an optional `preferredIndex` option; default `-1` preserves existing behavior for all other consumers (task page objective display).
- **Combined (grouped) view** (`useNeededItems.ts`): when searching, a pooled objective that matched an accepted item is grouped under the matched turn-in item (identified by name or short name, mirroring the search match) and `objectivesByItemId` registers it under the same key so the grouped modal lists the task. Without an accepted match, the primary item stays canonical under the grouped-view rule that nameless items are not grouped. Grouping stays single-count: the objective contributes to exactly one group, so re-keying never double-counts.
- **Grouped modal pool badge** (`NeededItemGroupedModal.vue`): objective rows now show the existing "Any of N" popover badge when the objective accepts multiple items, consistent with row/card views.
- **Docs** (`docs/SYSTEMS.md`): records the pooled-objective identity invariant under "When this doc is wrong" — progress stays objective-keyed while the displayed/grouped item identity is display-only, and the search filter, display pin, and grouped-view match must stay aligned.

No locale changes: the existing `needed_items.any_of_items_short` / `needed_items.any_of_items` keys are reused. No new runtime dependencies; no changes to data fetching or upstream shapes (uses the existing display-only `acceptedItems` metadata).

## Validation

Branch is rebased on latest `main` (HEAD `670694a9`); validated with a clean tree (only unstaged `.nuxtrc` tooling churn from `nuxt prepare`).

- Focused Vitest set (`useCyclingItem`, `useNeededItems`, `useNeededItemsRouteSync`, `useItemDistribution`, `neededitems` feature tests, `AcceptedItemsPopover`, tasks objective equipment/grouping tests) — 114 tests passed, including pooled search, pinning, short-name-only grouping, primary-fallback grouping, and pin edge cases.
- `pnpm run typecheck` — pass.
- `pnpm run lint` (design lint + eslint `--max-warnings=0` + blank lines) — pass.
- `pnpm run lint:fallow --base origin/main` — pass, new-only gate reports 0 introduced findings (search matchers and group-target resolution were split into small helpers to stay under the complexity budgets).
- `pnpm run i18n:check` — pass (no locale changes).
- CI on HEAD: all checks green, including Fallow audit, SonarCloud Quality Gate (0 new issues), CodeQL, Lighthouse, all 4 test shards, and Workers.
- Manual browser verification against local dev server with live json.tarkov.dev data: searching "Augmentin" (inline and via the `?q=Augmentin` deep link) on /needed-items returns 6 pooled objectives across list, grid, and combined views — grid pins the Augmentin image with "Any 52/56/3358" badges; combined view shows one "Augmentin antibiotic pills" group (0/233) whose modal lists Pets Won't Need It (Any 56) and the Building Foundations sell pools with per-row "Any N" badges; the accepted-items popover lists all 56 items; empty search still shows the full 854-item list.

## Review outcomes

- CodeRabbit: 1 finding (fallback-branch test coverage) — fixed and confirmed by the reviewer; thread resolved.
- cubic: 1× P2 (short-name grouping consistency) — fixed; 2× P3 (fallback test, `currentIndex` docs) — fixed/documented; re-review on HEAD passes with no findings; threads resolved.
- Greptile: 1× P2 (document the grouping-identity invariant) — addressed in `docs/SYSTEMS.md`; thread resolved.
- SonarCloud: Quality Gate passed, 0 new issues on HEAD (an initial S3776 cognitive-complexity finding on the new code was resolved by extracting the search matchers).

`docs/SYSTEMS.md` updated in the same change per the maintenance contract; no other system docs are affected.

### CODE REVIEW

File: app/composables/useNeededItems.ts [hunk 1]
```diff
@@ -531,20 +539,43 @@ export function useNeededItems(options: UseNeededItemsOptions = {}): UseNeededIt
...
+   */
+  const resolveGroupTarget = (
+    need: NeededItemTaskObjective | NeededItemHideoutModule
+  ): GroupTarget | null => {
+    const primaryData = getNeededItemData(need);
+    if (!primaryData?.id) return null;
+    if (need.needType === 'taskObjective' && search.value) {
+      const matchIndex = findAcceptedItemMatchIndex(need.acceptedItems, search.value);
+      const matchedItem = matchIndex >= 0 ? need.acceptedItems?.[matchIndex] : undefined;
+      if (matchedItem?.id && matchedItem.name) {
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/composables/useNeededItems.ts]: Fixed in 670694a9. The grouped-view accepted match now resolves the target through the same name-or-short-name matching as the search filter and display pin (toGroupTarget(findAcceptedGroupMatch(need))), so a short-name-only match groups and pins under the matched item consistently. The primary-item fallback keeps the existing grouped-view rule that nameless items are not grouped; the invariant is now documented in docs/SYSTEMS.md.

File: app/composables/__tests__/useNeededItems.test.ts [hunk 2]
```diff
@@ -555,6 +555,67 @@ describe('useNeededItems', () => {
...
+      expect(groupIds).toContain('item-augmentin');
+      expect(groupIds).not.toContain('item-cms');
+      const registered = neededItems.objectivesByItemId.value.get('item-augmentin');
+      expect(registered?.taskObjectives.map((objective) => objective.id)).toEqual(['obj-pool']);
+    });
+    it('keeps the primary item as the group key when no accepted item matches', async () => {
+      const { neededItems, search } = await setup({
+        metadataStore: { neededItemTaskObjectives: [createPooledObjective()] },
+      });
+      search.value = 'cms';
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/composables/__tests__/useNeededItems.test.ts]: Valid — fixed in 670694a9. The test now searches the task name ('task one'), which includes the objective via the task-name matcher without matching any accepted item, so the primary-item group fallback is actually exercised (and asserts the objective is registered under item-cms in objectivesByItemId). Added a separate test covering a short-name-only accepted match.

[the remark this answers is not in the document]
Comment (DysektAI) [reply app/composables/__tests__/useNeededItems.test.ts]: Fixed in 670694a9 — the test now searches 'Task One' (task-name match, no accepted-item match) and asserts the group key stays item-cms, exercising the primary-item fallback branch.

File: app/composables/useCyclingItem.ts [hunk 3]
```diff
@@ -72,6 +86,9 @@ export function useCyclingItem(
   );
   const currentItem = computed(() => {
     const fallback = toValue(primaryItem) ?? null;
+    if (pinnedIndex.value >= 0) {
+      return itemList.value[pinnedIndex.value] ?? fallback;
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/composables/useCyclingItem.ts]: Valid — addressed in 670694a9 by updating the contract docs rather than coupling the rotation index to the pin: currentIndex is documented as the rotation index (advances while cycling, resets when paused or pinned), while the displayed item when pinned is itemList[preferredIndex]. No consumer reads currentIndex today, so exposing the pin there would be unused surface; the docs now make the split explicit.

File: app/composables/useNeededItems.ts [hunk 4]
```diff
@@ -531,20 +539,43 @@ export function useNeededItems(options: UseNeededItemsOptions = {}): UseNeededIt
...
+        const { id, name } = matchedItem;
+        return { id, data: matchedItem, name };
```
[the remark this answers is not in the document]
Comment (DysektAI) [reply app/composables/useNeededItems.ts]: Done in 670694a9 — added a "Needed Items pooled objective search and grouping" subsection under "When this doc is wrong" in docs/SYSTEMS.md covering the two-identity model (objective-keyed progress vs display-only item identity), search-time re-keying, single-group contribution, and the requirement to keep findAcceptedItemMatchIndex and the grouped-view accepted match aligned.
